rfcs/0005-artifact-revision-and-digest/README.md

```diff
@@ -0,0 +1,352 @@
+# RFC-0005 Artifact `Revision` format and introduction of `Digest`
+
+**Status:** provisional
+
+<!--
+Status represents the current state of the RFC.
+Must be one of `provisional`, `implementable`, `implemented`, `deferred`, `rejected`, `withdrawn`, or `replaced`.
+-->
+
+**Creation date:** 2022-10-20
+
+**Last update:** 2022-10-24
+
+## Summary
+
+This RFC proposes to establish a canonical format for an `Artifact` which
+points to a specific checksum (e.g. an OCI manifest digest or Git commit SHA)
+of a named pointer (e.g. an OCI image tag or Git tag). In addition, it proposes
+to include the algorithm name (e.g. `sha256`) as a prefix to any advertised
+checksum in an `Artifact` and further referring to it as a `Digest` as opposed
+to a `Checksum`.
+
+## Motivation
+
+The current `Artifact` type's `Revision` field format is not "officially"
+standardized (albeit assumed throughout our code bases), and has mostly been
+derived from `GitRepository` which uses `/` as a delimiter between the named
+pointer (a Git branch or tag) and a specific (SHA-1, or theoretical SHA-256)
+revision.
+
+Since the introduction of `OCIRepository` and with the recent changes around
+`HelmChart` objects to allow the consumption of charts from OCI registries.
+This could be seen as outdated or confusing due to the format differing from
```

Comment:
```suggestion
`HelmChart` objects to allow the consumption of charts from OCI registries,
this could be seen as outdated or confusing due to the format differing from
```